sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
Ah, I agree with your analysis regarding the timezone-awareness of Date. I think we can solve the Timestamp inconsistency by unexporting `Date.Time` and only exposing a constructor that `UTC()`s incoming `time.Time`s; I think that would suffice for removing any perceived symmetry between the types. WDYT?